File: Behavioral_Design_Pattern/Observer Design Pattern/WithJava/WithObserverPattern/WeatherApp.java
package WithObserverPattern;

import WithObserverPattern.Observer.PhoneDisplay;
import WithObserverPattern.Observer.TVDisplay;
import WithObserverPattern.Subject.WeatherStation;

public class WeatherApp {
    public static void main(String[] args) {
        WeatherStation weatherStation = new WeatherStation();

        TVDisplay tvDisplay = new TVDisplay();
        PhoneDisplay phoneDisplay = new PhoneDisplay();

        weatherStation.addObserver(phoneDisplay);
        weatherStation.addObserver(tvDisplay);

        weatherStation.setWeather("Holocast");
        
    }
}

File: System Design Python Codes/Behavioral Design Pattern/Observer_Design_Pattern/With/WeatherApp.py
from Observer.PhoneDisplay import PhoneDisplay
from Observer.TVDisplay import TVDisplay
from Subject.WeatherStation import WeatherStation

def main():
    weatherStation1 = WeatherStation()
    
    ws1Devices = [PhoneDisplay(), TVDisplay()]
    
    for device in ws1Devices:
        weatherStation1.addObserver(device)
    
    weatherStation1.setWeather("Holocasty")

if __name__ == "__main__":
    main()